### PR TITLE
Add e2e tests for disable nodeport lb feature

### PR DESCRIPTION
**Pull request by aojea**
/kind feature

#### What this PR does / why we need it:

Add more testing for the feature to allows to disable nodeports on Load Balancer Services

https://github.com/kubernetes/enhancements/tree/master/keps/sig-network/1864-disable-lb-node-ports

```release-note
NONE
```

ref: https://github.com/kubernetes/enhancements/issues/1864

### CODE REVIEW

**Comment by aojea** [comment]
```
k8s.io/kubernetes/pkg/controller/statefulset: TestStatefulSetControl/CreatePodFailure/Burst/SetDeleteOnly/StatefulSetAutoDeletePVCEnabled expand_more	0s
k8s.io/kubernetes/pkg/controller/statefulset: TestStatefulSetControl expand_more	23s
k8s.io/kubernetes/pkg/kubelet/eviction: TestStart expand_more
```
unrelated
/test pull-kubernetes-unit

**Comment by aojea** [comment]
the ipvs job failure are sig-storage unrelated issues, is being worked out by sig-storage

**Comment by aojea** [comment]
the new e2e test is running and passing in this job
https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106522/pull-kubernetes-e2e-gci-gce-ingress/1461328276428427264

File: test/e2e/framework/service/jig.go [hunk 1]
```diff
@@ -474,7 +474,8 @@ func (j *TestJig) sanityCheckService(svc *v1.Service, svcType v1.ServiceType) (*
 	}
 
 	expectNodePorts := false
-	if svcType != v1.ServiceTypeClusterIP && svcType != v1.ServiceTypeExternalName {
+	if svcType != v1.ServiceTypeClusterIP && svcType != v1.ServiceTypeExternalName &&
```
Comment (thockin) [inline test/e2e/framework/service/jig.go:477]: I would flip this.

```
if svcType == v1.ServiceTypeNodePort || (svcType == v1.ServiceTypeLoadBalancer && getAllocateLoadBalancerNodePorts(svc)) {
 //...
```

and

```
func getAllocateLoadBalancerNodePorts(svc *v1.Service) bool {
    if svc.Spec.AllocateLoadBalancerNodePorts == nil {
        return true // back-compat
    }
    return *svc.Spec.AllocateLoadBalancerNodePorts
}

Comment (thockin) [reply test/e2e/framework/service/jig.go:477]: or even:

```
if needsNodePorts(svc) {
```

Those tiny helpers add a LOT to readability

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by thockin** [comment]
woops
/remove lgtm

File: test/e2e/framework/service/jig.go (hunk 1 above)
Comment (aojea) [reply test/e2e/framework/service/jig.go]: added helper func

**Comment by aojea** [comment]
> >
> k8s.io/kubernetes/pkg/controller/statefulset: TestStatefulSetControl/CreatePodFailure/Burst/Retain/StatefulSetAutoDeletePVCEnabled expand_more	0s
> k8s.io/kubernetes/pkg/controller/statefulset: TestStatefulSetControl expand_more
/test pull-kubernetes-unit

**Comment by aojea** [comment]
> Test_Run_OneVolumeDetachFailNodeWithReadWriteOnce

/test pull-kubernetes-unit

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**